java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -183,21 +187,23 @@ private void scheduleControllerTask(ShardedJobRunId jobId, IncrementalTaskId tas
       .param(TASK_ID_PARAM, taskId.toString());
     taskOptions.header("Host", settings.getTaskQueueTarget());
 
+    taskOptions.etaMillis(System.currentTimeMillis() + CONTROLLER_TASK_DELAY);
```

Comment:
added some delay here, bc again we're inside a txn, but enqueuing in the middle of it rather than at the end.